### DISCUSSION TITLE
Javascript object to python object conversion

### DISCUSSION THREAD

**Discussion by bbrenne** [pyodide/pyodide#4962] [Q&A]
Hi there,

I'm trying to get an access attributes of javascript objects into a python script using the notation obj.varname rather than through a dictionnary (dict['entryname']).

If I understood correctly, pyodide.toPy is supposed to convert object to object (without toPy it should convert to dict). So I did this test:

      const globals = pyodide.toPy({ x: 123, y: { a: 1, b: 2 } });
      pyodide.runPython(`
      print(x)
      print(y.a)`, { globals });

The output is the following:
123
AttributeError: 'dict' object has no attribute 'a'

So it looks like it's still converting to dictionnary, how can I make the conversion (if possible, without touching the python script) ?

Regards,

B

**Comment by Torsten85** [comment]
this works, but I don't know if its the right way.

```js
const globals = pyodide.toPy({})
globals.set('x', 123)
globals.set('y', { a: 1, b: 2 } )
```

**Reply by bbrenne** [reply]
Indeed it works! Thanks, that made my code cleaner.

**Comment by CNSeniorious000** [comment]
> pyodide.toPy is supposed to convert object to object (without toPy it should convert to dict)

No. `toPy` will turn a JavaScript `object` into a `PyProxy` of a Python `dict`. When you access it in python, it is a `dict`, otherwise it will be a `JsProxy` of the original JavaScript `object`.

See:

- [JavaScript to Python](https://pyodide.org/en/stable/usage/type-conversions.html#type-translations-jsproxy-to-py)
- [toPy](https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.toPy)
